src/struct_ls/_hypre_struct_ls.h

```diff
@@ -605,6 +603,13 @@ hypre_StructMatrix *hypre_SparseMSG2CreateRAPOp ( hypre_StructMatrix *R , hypre_
 HYPRE_Int hypre_SparseMSG2BuildRAPSym ( hypre_StructMatrix *A , hypre_StructMatrix *P , hypre_StructMatrix *R , HYPRE_Int cdir , hypre_Index cindex , hypre_Index cstride , hypre_Index stridePR , hypre_StructMatrix *RAP );
 HYPRE_Int hypre_SparseMSG2BuildRAPNoSym ( hypre_StructMatrix *A , hypre_StructMatrix *P , hypre_StructMatrix *R , HYPRE_Int cdir , hypre_Index cindex , hypre_Index cstride , hypre_Index stridePR , hypre_StructMatrix *RAP );
 
+/******************************************************************************
+ * Copyright 1998-2019 Lawrence Livermore National Security, LLC and other
+ * HYPRE Project Developers. See the top-level COPYRIGHT file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ ******************************************************************************/
+
```

Comment:
Why are the headers in this file?

Reply:
Are you asking about the "Copyright" disclaimer in this file? This file is now generated by `headers`

Reply:
Hi @liruipeng .  If you are introducing `headers` and `protos.h`, then you can remove the extra license header in `protos.h` and also all of the repeated prototypes for the capital `HYPRE_*` files that appear in `HYPRE_struct_ls.h`.  We can also do this later since this has been lying around for awhile already.  Thanks!

Reply:
@ulrikeyang @rfalgout I made the changes, removing extra license header and `HYPRE_*` prototypes in `protos.h`, also  adding some of the `HYPRE_*` prototypes into `HYPRE_struct_ls.h`. Please take another look. Thanks!

Reply:
Thanks!  -Rob